### PR TITLE
Update to Release 3

### PR DESCRIPTION
**Pull request by dionboles-asym**
This Release has
- Like operator support 
```
Q('ResearchSubject.primary_disease_type = "Lung%"')
q = Q('id LIKE "tcga-13-140%"')
```
 or using the query function 
```
 query('ResearchSubject.primary_disease_type LIKE "Lung%"')
```
- IN operator support 
```
Q("ResearchSubject.id IN ['4da7abaf-ac7a-41c0-8033-5780a398545c','010df72d-63d9-11e8-bcf1-0a2705229b82']")
```
or query function 

```
query(
        "ResearchSubject.id IN ['4da7abaf-ac7a-41c0-8033-5780a398545c','010df72d-63d9-11e8-bcf1-0a2705229b82']"
    )
```
- users can now create pandas dataframes from results using to_dataframe() 
```
  q1 = Q("ResearchSubject.Diagnosis.age_at_diagnosis > 50*365")
  q2 = Q('ResearchSubject.associated_project = "TCGA-OV"')
  q = q1.And(q2)
  r = q.run()
  r.to_dataframe().head()
```
- you can filter values in the run method and newly added files method
```
  q1 = Q("ResearchSubject.Diagnosis.age_at_diagnosis > 50*365")
  q2 = Q('ResearchSubject.associated_project = "TCGA-OV"')
  q = q1.And(q2)
  r = q.run(filter="id")
  r.to_dataframe().head()
```

### CODE REVIEW

File: cdapython/functions.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 from cda_client.model.query import Query
 from os import path
 from ssl import get_default_verify_paths
+from cdapython.ConvertionMap import convertionMap
```
Comment (fkaufman-asym) [inline cdapython/functions.py:5]: Should this be ConversionMap?

File: testColum.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from cdapython import columns
```
Comment (fkaufman-asym) [inline testColum.py:1]: Should the name of this file be nameColumn.py

File: cdapython/functions.py (hunk 1 above)
Comment (dionboles-asym) [reply cdapython/functions.py:5]: i'm doing a module import so that is the name of the file and i'm importing a dict or map into the functions file

Comment (dionboles-asym) [reply cdapython/functions.py:5]: Oh it was miss spelled lol

File: cdapython/constantVariables.py [hunk 3]
```diff
@@ -1,11 +1,15 @@
 # Versioning Year Month Day of last push
-VERSION = "2022.1.24"
-DATABASETABLE_VERSION = "all_v2_1"
-DATABASETABLE = "gdc-bq-sample.integration"
-CDA_API_URL_ENV = "https://cda.cda-dev.broadinstitute.org"
+VERSION: str = "2022.3.21"
+DATABASETABLE_VERSION: str = "all_v3_0_subjects_meta"
+DATABASETABLE: str = "gdc-bq-sample.dev"
```
Comment (druth314) [inline cdapython/constantVariables.py:4]: Are we continuing to use gdc-bq-sample as the project or are we switching it all to broad-cda-dev? It doesn't matter much to me but I just want to try to put dev and integration under a similarly named project and dataset and stuff

File: tests/test_conversion_testing.py [hunk 4]
```diff
@@ -1,7 +1,14 @@
+from pandas import DataFrame
 from cdapython import Q
```
Comment (fkaufman-asym) [inline tests/test_conversion_testing.py:2]: Import DataFrame from pandas is repeated.

File: cdapython/constantVariables.py (hunk 3 above)
Comment (dionboles-asym) [reply cdapython/constantVariables.py:4]: for integration we are using these tables but when we release to Maine I will switch them

**Review by fkaufman-asym — CHANGES_REQUESTED** [review]

File: tests/test_conversion_testing.py (hunk 4 above)
Comment (dionboles-asym) [reply tests/test_conversion_testing.py:2]: i just removed it thanks

File: README.md [hunk 5]
```diff
@@ -332,6 +332,77 @@ UNNEST(ResearchSubject) AS _ResearchSubject,
...
+````
+
+in the cda-python folder create a virtual environment
+using 
+
+```python
+python3 -m venv venv
+```
+
+and activate the environment by using
```
Comment (druth314) [inline README.md:360]: I'd just add a : at the end of the line

**Review by fkaufman-asym — APPROVED** [review]

File: cdapython/_get_unnest_clause.py [hunk 6]
```diff
@@ -6,7 +6,11 @@
 
 # A.B.C.D.column ->
 # SELECT DISTINCT(_D.column) FROM TABLE, UNNEST(A) AS _A, UNNEST(_A.B) AS _B, UNNEST(_B.C) AS _C, UNNEST(_C.D) AS _D
-def _get_unnest_clause(col_name):
+
+from typing import Union, Tuple, List
+
+
+def _get_unnest_clause(col_name: str) -> Tuple[str, List[str]]:
```
Comment (druth314) [inline cdapython/_get_unnest_clause.py:13]: How does this handle the top level fields that are lists like subject_associated_project? I though searching in a list required getting the unnest clause?

File: tests/test_icd_drs.py [hunk 7]
```diff
@@ -8,5 +8,3 @@
 WHERE (_ResearchSubject.primary_disease_type = 'Adenomas and Adenocarcinomas')
```
Comment (druth314) [inline tests/test_icd_drs.py:8]: Is this meant to still query on the V2 table?

File: tests/test_unique_version.py [hunk 8]
```diff
@@ -0,0 +1,3 @@
+from cdapython import unique_terms
+
+print(unique_terms("vital_status", version="all_v2_1"))
```
Comment (druth314) [inline tests/test_unique_version.py:3]: Should we change the version on this or leave that to the testers?

**Pull request merged**